src/components/utils/useElementSize.ts

```diff
@@ -11,6 +11,10 @@ interface ElementSize {
 
 export function useElementSize<T extends HTMLElement = HTMLDivElement>(
     ref: MutableRefObject<T | null> | null,
+    // can be used, when it is needed to force reassign observer to element
+    // in order to get correct measures. might be related to below
+    // https://github.com/WICG/resize-observer/issues/65
```

Comment:
@ValeraS ResizeObserver not always firing (for example changing from 'm' to 'l' size it doesn't fire), so for such cases added `key` param (in case of `TextInput` size will be used as key)